src/solver/opamSolver.ml

```diff
@@ -167,15 +167,25 @@ let opam2cudf_map universe version_map packages =
     OpamPackage.Set.fold (fun nv -> OpamPackage.Map.add nv true)
       (packages %% set) OpamPackage.Map.empty
   in
+  let avoid_versions =
+    OpamStd.Option.default OpamPackage.Set.empty @@
+    OpamStd.List.assoc_opt String.equal "avoid-version" universe.u_attrs
+  in
   let base_map =
     OpamPackage.Set.fold (fun nv ->
+        let pkg_extra = [
+          OpamCudf.s_source, `String(OpamPackage.name_to_string nv);
+          OpamCudf.s_source_number, `String(OpamPackage.version_to_string nv);
+        ] in
+        let pkg_extra =
+          if OpamPackage.Set.mem nv avoid_versions then
+            (OpamCudf.s_avoid_version, `Bool true)::pkg_extra
```

Comment:
This change should not be necessary after https://github.com/ocaml-opam/opam-0install-solver/pull/37/files#r1697389138